incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiBindingConfig.java

```diff
@@ -0,0 +1,249 @@
+import io.aklivity.zilla.runtime.engine.internal.stream.NamespacedId;
+
+public final class OpenapiBindingConfig
+{
+    public final long id;
+    public final String name;
+    public final KindConfig kind;
+    public final OpenapiOptionsConfig options;
+    public final List<OpenapiRouteConfig> routes;
+    public final HttpHeaderHelper helper;
+
+    private final long defaultRouteId;
+    private final IntHashSet httpOrigins;
+    private final Long2LongHashMap resolvedIds;
+    private final Object2ObjectHashMap<Matcher, PathItem> paths;
+    private final Map<CharSequence, Function<PathItem, String>> resolversByMethod;
+
+    public OpenapiBindingConfig(
+        BindingConfig binding,
+        long defaultRouteId)
+    {
+        this.id = binding.id;
+        this.name = binding.name;
+        this.kind = binding.kind;
+        this.defaultRouteId = defaultRouteId;
+        this.options = OpenapiOptionsConfig.class.cast(binding.options);
+        this.paths = new Object2ObjectHashMap<>();
+        options.openapis.forEach(c -> c.openapi.paths.forEach((k, v) ->
+        {
+            String regex = k.replaceAll("\\{[^/]+}", "[^/]+");
+            regex = "^" + regex + "$";
+            Pattern pattern = Pattern.compile(regex);
+            paths.put(pattern.matcher(""), v);
+        }));
+
+        this.routes = binding.routes.stream().map(OpenapiRouteConfig::new).collect(toList());
+
+        this.resolvedIds = binding.composites.stream()
+            .map(c -> c.bindings)
+            .flatMap(List::stream)
+            .filter(b -> b.type.equals("http"))
+            .collect(of(
+                () -> new Long2LongHashMap(-1),
+                (m, r) -> m.put(0L, r.id), //TODO: populate proper apiId
+                (m, r) -> m,
+                IDENTITY_FINISH
+            ));
+
+        this.httpOrigins = binding.composites.stream()
+            .map(c -> c.bindings)
+            .flatMap(List::stream)
+            .filter(b -> b.type.equals("http"))
+            .map(b -> NamespacedId.namespaceId(b.id))
+            .collect(toCollection(IntHashSet::new));
+        this.helper = new HttpHeaderHelper();
+
+        this.resolversByMethod = new TreeMap<>(CharSequence::compare);
+        resolversByMethod.put("POST", o -> o.post != null ? o.post.operationId : null);
+        resolversByMethod.put("PUT", o -> o.put != null ? o.put.operationId : null);
+        resolversByMethod.put("GET", o -> o.get != null ? o.get.operationId : null);
+        resolversByMethod.put("DELETE", o -> o.delete != null ? o.delete.operationId : null);
+        resolversByMethod.put("OPTIONS", o -> o.options != null ? o.options.operationId : null);
+        resolversByMethod.put("HEAD", o -> o.head != null ? o.head.operationId : null);
+        resolversByMethod.put("PATCH", o -> o.patch != null ? o.patch.operationId : null);
+        resolversByMethod.put("TRACE", o -> o.post != null ? o.trace.operationId : null);
```

Comment:
In general, prefer object is fully initialized before assignment, and then made readonly.

```suggestion
        Map<CharSequence, Function<PathItem, String>> resolversByMethod = new TreeMap<>(CharSequence::compare);
        resolversByMethod.put("POST", o -> o.post != null ? o.post.operationId : null);
        resolversByMethod.put("PUT", o -> o.put != null ? o.put.operationId : null);
        resolversByMethod.put("GET", o -> o.get != null ? o.get.operationId : null);
        resolversByMethod.put("DELETE", o -> o.delete != null ? o.delete.operationId : null);
        resolversByMethod.put("OPTIONS", o -> o.options != null ? o.options.operationId : null);
        resolversByMethod.put("HEAD", o -> o.head != null ? o.head.operationId : null);
        resolversByMethod.put("PATCH", o -> o.patch != null ? o.patch.operationId : null);
        resolversByMethod.put("TRACE", o -> o.post != null ? o.trace.operationId : null);
        this.resolversByMethod = Collections.unmodifiableMap(resolversByMethod);
```